lib/benchee.ex

```diff
@@ -60,13 +62,23 @@ for {module, moduledoc} <- [{Benchee, elixir_doc}, {:benchee, erlang_doc}] do
     end
 
     defp output_results(suite = %{configuration: %{formatters: formatters}}) do
-      Enum.each formatters, fn(output_function) ->
+      {parallelizable, serial} = Enum.split_with(formatters, &is_module?/1)
+
+      Formatter.parallel_output(suite, parallelizable)
+      Enum.each serial, fn(output_function) ->
         output_function.(suite)
       end
 
       suite
     end
 
+    defp is_module?(formatter) do
```

Comment:
`is_atom` seems like a kind of dangerous way to detect if something is a module. How about this:

```elixir
  defp is_module(formatter) when is_atom(formatter) do
    attrs = formatter.module_info(:attributes)

    attrs
    |> Keyword.get(:behaviour, [])
    |> Enum.member?(Benchee.Formatter)
  rescue
    false
  end

  defp is_module(_), do: false
```

Every module should respond to that `module_info/1` function, and if someone passes in a bare atom then it'll raise an error. If that error is raised, then it's not a module, and if it isn't raised then we know we have a module and can check to see which behaviours it implements.

I have a feeling we might want to either rename this function or separate the behaviour checking out from the module checking, but this should be pretty reliable for us.

However, I'm also going to put something on the mailing list, since it seems to me like an `is_module/1` function would be super helpful!

Reply:
huh, I thought about implementing a more sophisticated check but my reasoning was basically "why should anyone pass an atom in and if they do, blows up relatively soon either way" - this might still be nicer though.

Regarding the ML, also thought about using `Code.ensure_loaded?` but thought that we shouldn't run into this problem as everything should be compiled etc/beforehand:)